drivers/adc/adc_nrfx_saadc.c

```diff
@@ -575,8 +575,8 @@ static void correct_single_ended(const struct adc_sequence *sequence, nrf_saadc_
 				} else if (*sample < 0) {
 					*sample = 0;
 				}
+				sample++;
```

Comment:
This pointer should be incremented for all selected channels, not only the single-ended ones.